src/nbl/video/IDescriptorPool.cpp

```diff
@@ -4,64 +4,166 @@
 namespace nbl::video
 {
 
-bool IDescriptorPool::createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
+uint32_t IDescriptorPool::createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
 {
-    core::vector<SDescriptorOffsets> descriptorOffsets(count);
+    core::vector<SDescriptorOffsets> descriptorOffsets;
+    descriptorOffsets.reserve(count);
+
+    system::ILogger* logger = nullptr;
+    {
+        auto debugCallback = getOriginDevice()->getPhysicalDevice()->getDebugCallback();
+        if (debugCallback)
+            logger = debugCallback->getLogger();
+    }
 
     for (uint32_t i = 0u; i < count; ++i)
     {
         if (!isCompatibleDevicewise(layouts[i]))
-            return false;
+        {
+            if (logger)
+                logger->log("Device-Incompatible descriptor set layout found at index %u. Sets for the layouts following this index will not be created.", system::ILogger::ELL_WARNING, i);
+
+            break;
+        }
         
-        descriptorOffsets[i] = allocateDescriptorOffsets(layouts[i]);
+        if (!allocateDescriptorOffsets(descriptorOffsets.emplace_back(), layouts[i]))
+        {
+            if (logger)
+                logger->log("Failed to allocate descriptor offsets in the pool's storage for descriptor set layout at index %u. Sets for the layouts following this index will not be created.", system::ILogger::ELL_WARNING, i);
+
+            descriptorOffsets.pop_back();
+            break;
+        }
+    }
+
+    const auto successCount = descriptorOffsets.size();
+    assert(count >= successCount);
+    std::fill_n(output + successCount, count - successCount, nullptr);
```

Comment:
if `createDescriptorSets_impl` fails, then the rest of `output` needs to be filled with `nullptr` this is why my code listing looked like this

```cpp
const bool creationSuccess =  createDescriptorSets_impl(descriptorOffsets.size(),layouts,descriptorOffsets.data(),output);
const auto successCount = creationSuccess ? descriptorOffsets.size():0u;
std::fill_n(output+successCount,count-successCount,nullptr);
if (!creationSuccess)
    cancelDescriptorOffsetsAllocations(std::move(descriptorOffsets));
return successCount;
```

Reply:
yes because you call `createDescriptorSets_impl` later and you only fill `output` with nullptr in the range `[successCount,count)`

and I don't see `createDescriptorSets_impl` filling its output with `nullptr` when it fails (and it shouldn't either as that would push error handling responsibility into the backends)

Reply:
Ah, you're right.
Resolved.